### PR TITLE
Attempt to justify and explain unevaluated keyword phrasing

### LINKED ISSUE #998: Defining "evaluation"
From @Relequestual:

> I've been thinking about this section and its phrasing.
It's the first time in the whole document we mention "evaluation", and we don't really define what it means.
>
> People (including myself previously) incorrectly assume that if a location has been evaluated (had something applied to it), then it counts as having been evaluated.
>
> I'm going to riff some ideas...
>
> We say here "successful evaluations", we mean if an application doesn't result in validation errors, and therefore the applied schema or keyword is correctly applicable to the instance location.
>
> For example, if you had a schema...
anyOf [ { title> unicycle, props > wheels > const > 1 }, { title: bicycle, props > wheels > const > 2 } ]
>
> If your instance was a unicycle, the first subschema item in the anyOf would be applied and successfully evaluated. Because it only has one wheel, we know it's a unicycle. It is not a bicycle, so that whole subschema is not successfully evaluated.
>
> The instance is successfully evaluated as (or determined to be) a unicycle.
>
> We can further justify this terminology by the fact that evaluation can be short-circuited, and therefore there's no expectation for a schema to be fully evaluated (although that would be a weak argument, and probably confuse things).
>
> We need to clearly define what successful evaluation means, and what unsuccessful evaluation means, with justification (IMHO). I'm not currently convinced I can do that.

### PR DESCRIPTION
**Pull request by Relequestual**
Fixes #998

### CODE REVIEW

**Comment by Relequestual** [comment]
I tried to take a stab at this, but I'm not sure if this actually helps, or causes further confusion.
Please give me your thoughts!

File: jsonschema-core.xml [hunk 1]
```diff
@@ -2576,6 +2576,14 @@
                 to contribute to whether or not the item or property has been evaluated.
                 Only successful evaluations are considered.
             </t>
+            <t>
+                If an item in an array or an object property is "successfully evaluated", it
+                is logically considered to be valid in terms of the representation of the
+                object or array that's expected. For example if a subschema represents a car,
+                which requires between 2-4 wheeles, and the value of "wheeles" is 6, the instance
```
Comment (handrews) [inline jsonschema-core.xml:2583]: Is "wheeles" British spelling of American "wheels"?  I was not aware of this one, so it's a bit jarring to read.

However, this explanation is fantastic and I enthusiastically approve of this PR.

**Review by handrews — APPROVED** [review]

**Review by ssilverman — CHANGES_REQUESTED** [review]
“wheeles” -> “wheels”

File: jsonschema-core.xml (hunk 1 above)
Comment (Relequestual) [reply jsonschema-core.xml:2583]: Sadly, this is just my almost dyslexia coming into play

File: jsonschema-core.xml [hunk 2]
```diff
@@ -2576,6 +2576,14 @@
...
+                object is not "evaluated" to be a car, and the "wheeles" property is considerd
```
Comment (karenetheridge) [inline jsonschema-core.xml:2584]:
```suggestion
                object is not "evaluated" to be a car, and the "wheeles" property is considered
```

**Review by karenetheridge — CHANGES_REQUESTED** [review]

**Review by gregsdennis — APPROVED** [review]

**Review by ssilverman — APPROVED** [review]

**Comment by Relequestual** [comment]
🚢

**Pull request merged**